### DISCUSSION TITLE
BeatBuddy Drum Kit

### DISCUSSION THREAD

**Discussion by NoAliby** [hydrogen-music/hydrogen#1420] [General]
I have compiled a drum kit that I use for creating loops for my BeatBuddy.

It has the same Drum Instrument Names as the common BeatBuddy Kits (Brushes, Dance, Metal, Rock, Standard) and works for Jazz as well (the only differences are that the jazz kit uses a cowbell for the splash and a stick-on-stick shot instead of a Hand clap)

The Midi mapping is according to General Midi standards so that the export to midi is correct and can be loaded into the BeatBuddy Manager with the correct instruments assigned.

Let me know if this helps you or what you think.

[Updated File Below]

**Comment by theGreatWhiteShark** [comment]
Hey @NoAliby ,

I have a couple of questions:

1. are the samples you used part of the BeatBuddy and how are they licensed?
2. how come you didn't use the GMRockKit? If you require a General MIDI compatible kit this default you should do the job as well. Or was something not working there?

**Comment by NoAliby** [comment]
Hi @theGreatWhiteShark 

The samples are all from current free Hydrogen Kits (actually mostly from the GMRockKit, except things like the tambourine and shaker from either "ErnysPercussion" of "Gimme a Hand 1.0")

The GMRockKit is "almost" General Midi compatible. The Cowbell, Ride and Ride2 are placed on Midi Notes for Toms. Also the BeatBuddy excludes samples on some effectively duplicate"sounds so those "notes" are missing, plus there are a few more sounds included on the BeatBuddy standard kits. Before I would use the GMRockKIt and then when you import the midi to the BeatBuddy Manager you just move "unsupported "  sounds to the appropriate Tom etc. But having a kit with the correct layout and midi notes assigned just makes it far simpler. (Now if I can only get one of the midi2hydogen import tools working this will be perfect)

Here is a table comparing GM with the GMRockKit and The BeatBuddy standard kit.

MIDI Note | GMRockKit | GM Standard | My BeatBuddy
-- | -- | -- | --
36 | Kick | Bass Drum 1 | Kick Drum
37 | Stick | Side Stick/Rimshot | Cross Stick
38 | Snare | Snare Drum 1 | Snare
39 | Hand Clap | Hand Clap | Handclaps
40 | Snare Rimshot | Snare Drum 2 |  
41 | Floor Tom | Low Tom 2 |  
42 | Hat Closed | Closed Hi-hat | Hi-Hats Closed
43 | Tom 2 | Low Tom 1 | Tom 4
44 | Hat Pedal | Pedal Hi-hat | Foot Hi-Hat
45 | Tom 1 | Mid Tom 2 | Tom 3
46 | Hat Open | Open Hi-hat | Hi-Hat Open
47 | Cowbell     (not GM) | Mid Tom 1 |  
48 | Ride       (not GM) | High Tom 2 | Tom 2
49 | Crash | Crash Cymbal 1 | Crash Cymbal 1
50 | Ride 2      (not GM) | High Tom 1 | Tom 1
51 | Splash | Ride Cymbal 1 | Ride Cymbal
52 | Hat Semi-Open | Chinese Cymbal |  
53 | Bell | Ride Bell | Ride C. Bell
54 |   | Tambourine | Tambourine
55 |   | Splash Cymbal | Splash 1
56 |   | Cowbell | Cowbell
57 |   | Crash Cymbal 2 | Crash Cymbal 2
59 |   | Ride Cymbal 2 | Splash 2
82 |   | Shaker (GM2) | Shaker


</body>

</html>

**Comment by NoAliby** [comment]
I have updated the file to capture the Sample Sources as shown below. 

Edited GMRockKit Samples (Alistair Baty) GPL
  - Tom 2 (Pitch Change on Tom 2)
  - Tom 4 (Pitch Change on Floor Tom)
  - Cymbal 2 (Pitch Change on Cymbal)
  - Splash 2 (Pitch Change on Splash)
Gimme A Hand 1.0 (Glen MacArthur) GPL
  - Hand Clap
  - Tambourine
ErnysPercussion (Erny) "Have Fun!"
  - Cabasa (Down, Up and Cut as Shaker Layers)
GMRockKit (Glen MacArthur / Sebastian Moors) GPL
  - All Other Drums

(also some clean-up of unused samples)
[See v1.0 release below]

**Comment by theGreatWhiteShark** [comment]
Nice. Thanks!

We might add this kit to the others available for download within Hydrogen. Would it be OK for you?

> (Now if I can only get one of the midi2hydogen import tools working this will be perfect)

Have you tried using another software capable of reading and sending MIDI? You could wire it's output to the MIDI input of Hydrogen and record the incoming notes. (But I haven't done it myself. Not sure whether this yield satisfiable results)

**Reply by NoAliby** [reply]
I made a few changers to the drum kit last night.
I'll upload the new one and then you are welcome to share it in the downloads.

I will see if I can use a midi program to stream it to hydrogen - haven't tried that approach  yet

**Reply by NoAliby** [reply]
@theGreatWhiteShark 
I have uploaded v1.0, which you are welcome to add to the downloads.
It will be helpful if we can use this thread for others to add comments or requests.
_PS. I also found that I could not install kits from inside Hydrogen (Online Import). I had to download the kits from SourceForge and then install them using the Local File option._

**Reply by theGreatWhiteShark** [reply]
> PS. I also found that I could not install kits from inside Hydrogen (Online Import). I had to download the kits from SourceForge and then install them using the Local File option.

Could you create an issue regarding this matter and provide some more details, like your OS, Hydrogen version, debug log? This sounds like a bug.

**Reply by NoAliby** [reply]
@theGreatWhiteShark where do I find the debug logs

**Reply by theGreatWhiteShark** [reply]
If you are using Linux or OSX just open a terminal and enter `hydrogen -VDebug`. The log will appear in the terminal

If you are using Windows, open the command interface and enter `start "" "C:\Program Files\Hydrogen\hydrogen.exe" -VDebug`. Afterwards, use the main menu Debug > Open Debug Log.

**Reply by NoAliby** [reply]
Thanks.
Issue Logged 
https://github.com/hydrogen-music/hydrogen-music/issues/26

**Comment by NoAliby** [comment]
v1.0 of My BeatBuddy Kit

Please add comments about the kit and any issues or proposed changes.
Things like imbalanced volume levels or filtering requirements etc. or if there is any issues when importing into the BeatBuddy Manager.

[My BeatBuddy Kit v1.0.zip](https://github.com/hydrogen-music/hydrogen/files/7919752/My.BeatBuddy.Kit.v1.0.zip)

**Comment by NoAliby** [comment]
[My BeatBuddy Kit v1.1.zip](https://github.com/hydrogen-music/hydrogen/files/7962926/My.BeatBuddy.Kit.v1.1.zip)

V1.1 
- Fix Tom 1 Pitch
- BVP saturation corrections
- Add Panning mods

**Comment by NoAliby** [comment]
[My BeatBuddy Kit v1.2.zip](https://github.com/hydrogen-music/hydrogen/files/8457387/My.BeatBuddy.Kit.v1.2.zip)

V1.2 

- Fix Patching of Tom 4 to Note 43
- Pitch on Tom1=+5 ,  Tom 3=+1

**Reply by theGreatWhiteShark** [reply]
Thanks a lot!

Sorry for not adding the drumkit to our online library earlier. I was quite busy.

I noticed two things that are slightly off in your drumkit and they are both our fault :D (so, thanks for pointing me to them too). Firstly, the `pan` is stored in a way the overall drumkit will be invalid in Hydrogen versions < 1.2 and, secondly, in one instance the `pitchOffset` is not 0 but `-3.65078e-08`. Both things I have already fixed in `master`. Could you export the kit again to cover them?

FYI: our current implementation of the sound library/online import does not support any version control. In case you make a newer version of the drumkit there are the options of a) using it to replace the former one or b) offering two versions next to each other. But in most cases a) is more favorable.

**Reply by NoAliby** [reply]
@theGreatWhiteShark 
Could you give me some specific instructions on fixing the two issues?
1. I just used the Mixer to set the pan in the drum kit. How do I fix it so it is compatible with older Hydrogen versions?
2. Do I manually go though and zero Pitch offset, or edit it in the XML file? There were only two instruments that use it.

For versions I have been doing the following, assuming people will over write previous versions
- adding the version number on the zip file
- Including the version in the description text. Both at the top and in the version history at the bottom.

**Reply by theGreatWhiteShark** [reply]
> 1. I just used the Mixer to set the pan in the drum kit. How do I fix it so it is compatible with older Hydrogen versions?

It's enough to upgrade to the latest version on `master` and to export it again. Which OS you are working with? For MacOS or Windows I can prepare the latest executable for you. On Linux you can use the `build.sh mm x` script to build it from source.

 > 2. Do I manually go though and zero Pitch offset, or edit it in the XML file? There were only two instruments that use it.

Inside the latest version you can update the value by setting it again in the mixer (e.g. clicking the rotary and typing `0.0` using your keyboard). But as there is just a single value that needs to be adjusted, it's probably faster to edit it in the XML file right away.

About the name: do you want to keep the "My " prefix when making it public?

**Reply by NoAliby** [reply]
I am running Windows 10. If you can link me to the latest  master  build that would be great!
We can drop the "My". Initially, I wanted to differentiate it from the actual BeatBuddy samples, in case of copy right. But I have posted on their site and with no opposition from Singular Sound to using the actual samples I will do so in the future as V2.0. 
I will rename it to "BeatBuddy Drum Kit" on the next export as v1.3

**Reply by theGreatWhiteShark** [reply]
> If you can link me to the latest master build that would be great!

[Hydrogen-1.1.1-win64.exe](https://ci.appveyor.com/api/buildjobs/t4nwtfyxxer59d9u/artifacts/Hydrogen-1.1.1-win64.exe) and [Hydrogen-1.1.1-win32.exe](https://ci.appveyor.com/api/buildjobs/rgu153xjjls79c7s/artifacts/Hydrogen-1.1.1-win32.exe)

**Reply by NoAliby** [reply]
I think I have covered all that. Saved a new V1.3.

_One thing with the latest "1.1.1" is that I can't seem to change the "DefaultSong.h2song" like I used to. I created one with specific pattern names in the song and then linked to my own Drum Kit. Before if I made a new song by clicking "NEW", it would load that default ".h2song". Now it load GMRockKit and the song patterns 1 thru 10. Any ideas?_

**Comment by NoAliby** [comment]
[BeatBuddy Kit v1.3.zip](https://github.com/hydrogen-music/hydrogen/files/8568994/BeatBuddy.Kit.v1.3.zip)

V1.3 

- Renamed to BeatBuddy Kit (No "MY"), 
- Kick Pitch=0, 
- Fixed Panning

**Reply by theGreatWhiteShark** [reply]
Hey.

While uploading I noticed one last thing. The current license of the drumkit is GPL. However, this license is more suitable to code than to creative output. How about changing it to [CC-BY-SA](https://creativecommons.org/licenses/by-sa/3.0/)? This is somewhat the GPL of the cultural and artistic world.

**Reply by NoAliby** [reply]
That does makes sense.
But as the samples in this compilation are based on other kits that use GPL it would be appropriate to be consistent with those.

My BeatBuddy Kit **GPL**
... using samples from:
- Gimme A Hand 1.0 (Glen MacArthur) **GPL**
- GMRockKit (Glen MacArthur / Sebastian Moors) **GPL**
- ErnysPercussion (Erny) "Have Fun!"

**Reply by theGreatWhiteShark** [reply]
Hmm. Your right. I wasn't aware of this. Especially that we host drumkits that not have been licensed at all (Ernys).

Okay. I will upload it with the GPL one.

**Reply by theGreatWhiteShark** [reply]
Done. You should be able to download it via the Online Import in Hydrogen after hitting "Update list" in the download dialog.
